WordPress/gutenberg
# Pattern block: Add experimental flag and syncStatus attrib to allow testing of partial syncing

## What?
If the syncStatus attribute of a pattern is set to `partial` a pattern block wrapper is added and and templateLock is set to `contentOnly`

## Why?

- Lays the groundwork for allowing the original pattern structure to be update and applied to the content of the pattern instance
- Allows for the option to shuffle pattern content with similar patterns

See https://github.com/WordPress/gutenberg/discussions/50456 for some background discussion.

## How?

If pattern has `syncStatus===partial` adds a parent div that wraps the content of the pattern. Also currently defaults the innerBlocks templateLock to `contentOnly`.

This is obviously going to change quite a bit in order to get partial syncing actually working as [discussed here](https://github.com/WordPress/gutenberg/discussions/50456), but it at least gives us a starting point of having Pattern block wrapper left in the post content within which to work.

**Important notes**: 

- a partially synced pattern block could impact the overall alignment of the innerBlocks as some alignments depend on the alignment settings of the parent block, and the partially synced block inserts a new parent wrapper. This will be resolved in a follow up PR
- In this PR some code was mistakenly left in the frontend render code that caused any pattern blocks with no `syncStatus` attrib set to still get wrapped in a pattern block. This was fixed in #50646 

## Testing Instructions

Prerequisite: enable the Enhanced Patterns experiment toggle in the Gutenberg plugins experiment screen.

- With TT3 theme in editor code view add ``
- Switch to editor view and make sure the pattern is added as currently in trunk, ie. pattern block is removed and inner blocks of pattern added with no wrapper
- Switch to code view and add `` 
- Switch back to editor view and note the CTA pattern has a `Pattern` block wrapper around it, editing is set to contentOnly, but it does not set the `wide` alignment of its inner blocks on the pattern block itself.
- Also check display is correct in frontend (the pattern block wrapper will not have any alignment applied to it)

## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/3629020/1f2d0e1e-d741-423f-8188-43e1504ec2cc
